### PR TITLE
Test for duplicate global var handling

### PR DESCRIPTION
**Pull request by kjayapra-amd**
Adding test cases for handling test cases where there is a duplicate global var between modules.

### CODE REVIEW

File: tests/src/runtimeApi/module/hipModuleGlobalDup.cpp [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+/* HIT_START
+ * BUILD_CMD: global_kernel.code %hc --genco %S/global_kernel.cpp -o global_kernel.code
```
Comment (gargrahul) [inline tests/src/runtimeApi/module/hipModuleGlobalDup.cpp]: Change global_kernel to dyn_global_kernel1
Also add instruction for dyn_global_kernel2.code

Comment (kjayapra-amd) [reply tests/src/runtimeApi/module/hipModuleGlobalDup.cpp]: VDI uses Makefile, so did not notice an issue there. Updated the test file to reflect the correct test commands.

**Comment by mangupta** [comment]
Closing all existing pull requests that do not target the develop branch. In case this pull request is still valid, please rebase your changes against develop branch and resubmit.
